yarn-project/bb-prover/src/prover/bb_prover_cache.ts

```diff
@@ -0,0 +1,71 @@
+import { TxExecutionRequest } from "@aztec/circuit-types";
+import { Fr, PrivateKernelTailCircuitPublicInputs, RecursiveProof, VerificationKeyAsFields, VerificationKeyData } from "@aztec/circuits.js";
+import { CacheableExecutionResult } from "@aztec/simulator";
+
+import * as crypto from "crypto";
+import * as fs from "fs/promises";
+import * as os from "os";
+import path from "path";
+
+import { ClassConverter, type ClassConverterInput } from "../../../foundation/src/json-rpc/class_converter.js";
+import { convertFromJsonBuffer, convertToJsonBuffer } from "../../../foundation/src/json-rpc/convert.js";
+
+const SHOULD_USE_PROVER_CACHE = !!process.env.PROVER_CACHE || !!process.env.PROVER_CACHE_DIR;
+const PROVER_CACHE_DIR = process.env.PROVER_CACHE_DIR || path.join(os.homedir(), '.aztec', 'proof-cache');
```

Comment:
WDYT about using a gitignored folder local to the repo, rather than polluting user's HOME? Especially since this is a feature specific for devs.